### PR TITLE
[json-rpc] CORS: allowing all origins for the interface

### PR DESCRIPTION
**Pull request by mimoo**

### CODE REVIEW

**Comment by libra-action** [comment]
**This PR may have modified JSON-RPC server code.**

Breaking changes policy:

1. Chaning JSON-RPC method signatures or removing/modifying fields
   in the response in /V1 will not be accepted.
2. Adding fields is ok.
3. Adding a new JSON-RPC method is ok.

Please ensure you have also done the following to update the docs:

1. Update json-rpc/API-CHANGELOG.md.
2. Add/update type/method documentation under json-rpc/docs.

**Comment by mimoo** [comment]
@xli should this be documented in the changelog?

**Comment by rustielin** [comment]
How do we verify that this works? I just built an image from this PR and deployed to a testnet: http://rustielin.aws.hlw3truzy4ls.com/

Curl is not returning the right headers I think:

```
$ ▶ curl -d '{"jsonrpc": "2.0", "method": "get_account", "params": ["000000000000000000000000000000dd"]}' -H 'Content-Type: application/json' -i --verbose http://rustielin.aws.hlw3truzy4ls.com/
*   Trying 64:ff9b::2ce6:4e4e...
* TCP_NODELAY set
* Connected to rustielin.aws.hlw3truzy4ls.com (64:ff9b::2ce6:4e4e) port 80 (#0)
> POST / HTTP/1.1
> Host: rustielin.aws.hlw3truzy4ls.com
> User-Agent: curl/7.64.1
> Accept: */*
> Content-Type: application/json
> Content-Length: 91
>
* upload completely sent off: 91 out of 91 bytes
< HTTP/1.1 200 OK
HTTP/1.1 200 OK
< Date: Wed, 30 Sep 2020 20:30:57 GMT
Date: Wed, 30 Sep 2020 20:30:57 GMT
< Content-Type: application/json
Content-Type: application/json
< Content-Length: 1159
Content-Length: 1159
< Connection: keep-alive
Connection: keep-alive

<
* Connection #0 to host rustielin.aws.hlw3truzy4ls.com left intact
{"libra_chain_id":3,"libra_ledger_version":486588,"libra_ledger_timestampusec":1601497857028863,"jsonrpc":"2.0","id":null,"result":{"address":"000000000000000000000000000000dd","authentication_key":"cd907006959c0f4afb78568710e418ba3630930522a40bd3146f71de5b4f23d0","balances":[{"amount":4611686016239387903,"currency":"Coin1"},{"amount":4611686018427387903,"currency":"Coin2"},{"amount":9223371981554775807,"currency":"LBR"}],"delegated_key_rotation_capability":false,"delegated_withdrawal_capability":false,"is_frozen":false,"received_events_key":"0300000000000000000000000000000000000000000000dd","role":{"base_url":"","base_url_rotation_events_key":"0200000000000000000000000000000000000000000000dd","compliance_key":"","compliance_key_rotation_events_key":"0100000000000000000000000000000000000000000000dd","expiration_time":18446744073709551615,"human_name":"moneybags","preburn_balances":[{"amount":0,"currency":"Coin1"},{"amount":0,"currency":"Coin2"}],"received_mint_events_key":"0000000000000000000000000000000000000000000000dd","type":"designated_dealer"},"sent_events_key":"0400000000000000000000000000000000000000000000dd","sequence_number":3289}}* Closing connection 0
```

**Comment by mimoo** [comment]
that would be a good way to test it! It's been years since I read anything on CORS, but IIRC you should only get it when you use some verbs like OPTION and POST?

**Comment by xli** [comment]
I don't think api-changelog need to be updated.

File: json-rpc/src/tests/unit_tests.rs [hunk 1]
```diff
@@ -1306,7 +1324,7 @@ fn create_db_and_runtime() -> (
 ) {
     let mock_db = mock_db();
 
-    let host = "0.0.0.0";
+    let host = "127.0.0.1";
```
Comment (mimoo) [inline json-rpc/src/tests/unit_tests.rs:1327]: @xli is this OK o.o?

**Comment by mimoo** [comment]
added a test, it should work! If you're testing that from the browser, try:

```
var xhr = new XMLHttpRequest();
xhr.open("POST", "https://rustielin.aws.hlw3truzy4ls.com", true);
xhr.setRequestHeader('Content-Type', 'application/json');
xhr.send(JSON.stringify({
  "jsonrpc": "2.0", 
  "method": "get_account", 
  "params": ["000000000000000000000000000000dd"]
}));
```

File: json-rpc/src/tests/unit_tests.rs (hunk 1 above)
Comment (xli) [reply json-rpc/src/tests/unit_tests.rs:1327]: as long as test pass

**Review by xli — APPROVED** [review]

**Pull request closed without merging**